app/models/transaction_type.rb

```diff
@@ -30,7 +30,10 @@ class TransactionType < ApplicationRecord
   belongs_to :target_category, class_name: 'Entity', foreign_key: :target_category_id, optional: true
   belongs_to :expense_category, class_name: 'Entity', foreign_key: :expense_category_id, optional: true
 
+  validates :mode, presence: true
   validates :name, presence: true, uniqueness: true
+  validates :source_category_id, presence: true
+  validates :target_category_id, presence: true
```

Comment:
This validation enforces for all.  I think we should allow empty source and target for `name: 'Initialize'`.

Reply:
What you can do is to add 

```
validate :abc_validator

...

private

def abc_validator
   ...
end
```